packages/core-data/src/resolvers.js

```diff
@@ -228,7 +228,23 @@ export const getEntityRecords =
 				...query,
 			} );
 
-			let records = Object.values( await apiFetch( { path } ) );
+			let records, meta;
+			if ( entityConfig.supportsPagination && query.per_page !== -1 ) {
+				const response = await apiFetch( { path, parse: false } );
+				records = Object.values( await response.json() );
+				meta = {
```

Comment:
Would we need error handling here, like we do inside `apiFetch` when we parse the response?

Reply:
Well, if we had proper error handling (like storing the error in the state and status) probably but right now, I don't think there's anything special happening even in the "parse: true" case. The error is just ignored it seems (the finally is triggered obviously).